pypeit/pypeit.py

```diff
@@ -902,6 +901,11 @@ def extract_one(self, frames, det, sciImg, global_sky, sobjs_obj, skymask):
         spec_flex_table['spat_id'] = self.caliBrate.slits.spat_id
         spec_flex_table['sci_spec_flexure'] = self.redux.slitshift
 
+        # pull out maskdef_designtab from caliBrate.slits
```

Comment:
These lines seem peculiar. Why is this being done? Your comment does not explain why.

Reply:
@kbwestfall I think you understand how these data models work. Can you please provide some help here. The problem is the maskdef_designtab is somehow not be written in the slits object, even thought it is an attribute there. For that reason @deborape is making this an explicit argument to the data model and then adding it explicitly. But the slits are already part of the data model so why don't nested data models work.